From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Raine Makelainen <raine.makelainen@jolla.com>
Date: Mon, 28 Sep 2020 16:19:17 +0300
Subject: [PATCH] [sailfishos][qt] Initialize FreeType library properly.
 JB#51369

Previously GetFTLibrary() of the gfxPlatform was always returning
null making the font loading to fail.

Use MOZ_LOG="userfonts:5" to log user fonts.

Signed-off-by: Raine Makelainen <raine.makelainen@jolla.com>
---
 gfx/thebes/gfxQtPlatform.cpp | 6 ++++++
 gfx/thebes/gfxQtPlatform.h   | 2 ++
 2 files changed, 8 insertions(+)

diff --git a/gfx/thebes/gfxQtPlatform.cpp b/gfx/thebes/gfxQtPlatform.cpp
index 191fa494b903..035891d7240d 100644
--- a/gfx/thebes/gfxQtPlatform.cpp
+++ b/gfx/thebes/gfxQtPlatform.cpp
@@ -138,6 +138,12 @@ gfxQtPlatform::MakePlatformFont(const nsAString& aFontName,
                                  aLength);
 }
 
+FT_Library
+gfxQtPlatform::GetFTLibrary()
+{
+    return gfxFcPlatformFontList::GetFTLibrary();
+}
+
 void
 gfxQtPlatform::GetPlatformCMSOutputProfile(void *&mem, size_t &size)
 {
diff --git a/gfx/thebes/gfxQtPlatform.h b/gfx/thebes/gfxQtPlatform.h
index 928e9a33e123..f4490930a3da 100644
--- a/gfx/thebes/gfxQtPlatform.h
+++ b/gfx/thebes/gfxQtPlatform.h
@@ -66,6 +66,8 @@ public:
                                            const uint8_t* aFontData,
                                            uint32_t aLength) override;
 
+    FT_Library GetFTLibrary() override;
+
     static int32_t GetDPI();
 
     virtual gfxImageFormat GetOffscreenFormat() override;
-- 
2.26.2

